internal/pkg/scaffold/olm-catalog/csv_updaters.go

```diff
@@ -248,32 +249,45 @@ func (store *updaterStore) AddOwnedCRD(yamlDoc []byte) error {
 	if err := yaml.Unmarshal(yamlDoc, crd); err != nil {
 		return err
 	}
-	store.crds.Owned = append(store.crds.Owned, olmapiv1alpha1.CRDDescription{
+	crdDesc := olmapiv1alpha1.CRDDescription{
 		Name:    crd.ObjectMeta.Name,
 		Version: crd.Spec.Version,
 		Kind:    crd.Spec.Names.Kind,
-	})
-	store.crds.crKinds[crd.Spec.Names.Kind] = struct{}{}
+	}
+	store.crds.crKinds[crdDesc.Kind] = struct{}{}
+	// Parse CRD descriptors from source code comments and annotations.
+	gvk := schema.GroupVersionKind{Group: crd.Spec.Group, Version: crdDesc.Version, Kind: crdDesc.Kind}
+	if err := setCRDDescriptorForGVK(&crdDesc, gvk); err != nil {
+		return err
```

Comment:
Wrap this with more context since there are a few very specific errors being returned from the function.
```suggestion
		return errors.Wrapf(err, "failed to set CRD descriptors for %s", gvk)
```